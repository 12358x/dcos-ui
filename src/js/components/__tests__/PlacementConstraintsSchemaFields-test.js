/* eslint-disable no-unused-vars */
var React = require("react");
/* eslint-enable no-unused-vars */

const renderer = require("react-test-renderer");

const PlacementConstraintsSchemaFields = require("../PlacementConstraintsSchemaField")
  .default;
const PlacementValidators = require("#PLUGINS/services/src/js/validators/PlacementsValidators.js");

describe("PlacementConstraintsSchemaField", function() {
  describe("with valid content", function() {
    beforeEach(function() {
      PlacementValidators.validateNoBatchError = () => true;
    });

    it("displays edit constraints area", function() {
      const fieldProps = {
        formData: "[['hostname', 'MAX_PER', '1']]",
        disabled: false,
        name: "placement_constraints"
      };
      this.instance = renderer.create(
        <PlacementConstraintsSchemaFields
          fieldProps={fieldProps}
          label="label"
          schema={{}}
        />
      );

      var tree = this.instance.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe("with invalid content", function() {
    beforeEach(function() {
      PlacementValidators.validateNoBatchError = () => false;
    });

    it("displays form to edit constraints", function() {
      const fieldProps = {
        formData: "['hostname', 'MAX_PER', '2']",
        disabled: false,
        name: "placement_constraints"
      };

      this.instance = renderer.create(
        <PlacementConstraintsSchemaFields
          fieldProps={fieldProps}
          label="label"
          schema={{}}
        />
      );

      var tree = this.instance.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
