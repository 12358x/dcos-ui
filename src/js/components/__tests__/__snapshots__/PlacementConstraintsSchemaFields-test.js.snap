exports[`PlacementConstraintsSchemaField with invalid content displays form to edit constraints 1`] = `
<div>
  <label
    className="">
    label
  </label>
  <pre>
    {
      \"placement_constraints\": \"[\'hostname\', \'MAX_PER\', \'2\']\"
    }
  </pre>
  <p
    className="form-control-feedback">
    Unable to edit 
    placement_constraints
  </p>
</div>
`;

exports[`PlacementConstraintsSchemaField with valid content displays edit constraints area 1`] = `
<div>
  <label
    className="">
    label
  </label>
  <div
    onChange={[Function]}>
    <div>
      <div
        className="form-row row">
        <div
          className="form-group column-12">
          <a
            className="button button-primary-link button-flush"
            onClick={[Function]}>
            <svg
              className="icon icon-purple icon-tiny">
              <use
                xlinkHref="#icon-system--plus" />
            </svg>
            <span>
              Add Placement Constraint
            </span>
          </a>
        </div>
      </div>
    </div>
  </div>
  <p
    className="form-control-feedback" />
</div>
`;
